modules/tensor_mechanics/doc/content/documentation/systems/Modules/TensorMechanics/GeneralizedPlaneStrain/index.md

```diff
@@ -1,38 +1,44 @@
-# GeneralizedPlaneStrain Action
+# Generalized Plane Strain Action System
 
-Sets up a generalized plane strain model. This action can be generated using the [TensorMechanics Master action](../Master). A detailed description of generalized plane strain model can be found in the [formulation](tensor_mechanics/generalized_plane_strain.md) page.
+!syntax description /Modules/TensorMechanics/GeneralizedPlaneStrain/GeneralizedPlaneStrainAction
 
-## Example Input File Syntax
+## Description
 
-!listing modules/tensor_mechanics/test/tests/generalized_plane_strain/generalized_plane_strain_squares.i block=Modules/TensorMechanics/GeneralizedPlaneStrain
+This action sets up a generalized plane strain model. This action can be generated using the [TensorMechanics Master action](../Master). A detailed description of generalized plane strain model can be found in the [formulation](tensor_mechanics/generalized_plane_strain.md) page.
+
+!alert warning title=For 2D and 1D Simulations
+GeneralizedPlaneStrain only works for 1D axisymmetric or 2D generalized plane strain cases where the number of displacement variables is only one or two displacements.
+
+## Constructed MooseObjects
+
+The `GeneralizedPlaneStrain` Action is used to construct the kernels, strain materials, displacement variables, and user objects in a consistent manner as required for a generalized plane strain simulation.
```

Comment:
This action doesn't setup the strain materials, displacement variables

Reply:
Action documentation is a bit tricky because we hid so much of the normal inheritance, ect. from the user. I had in mind writing the action documentation as a 'one-stop shop' for the reader/user: I wanted to put all of the information that a user would need to completely use the GeneralizedPlaneStrain action on this single page. This way, the user won't need to go to the tensor mechanics master action page to find out which parameters to add for displacements and strain formulations. As I'm looking at this again, I realize that I can't then cheat with sending the user to tensor mechanics master action for auxkernels and auxvariables either--this documentation information should be on the gps page if I include the displacements and strains information.

@bwspenc @novasr How did each of you envision the documentation for the gps action? Do you see this documentation as a self contained 'here's everything you need to know about the gps action'? Or would you prefer that the gps action page has just what the gps adds onto the master action page and that we send users to the master action page for information about adding variables, strain formulations, and auxvariables/auxkernels? I'll address @hchen139's comments based on if inclusive or specialized gps action make more sense

Thanks for bringing up this point @hchen139!